tests/fixtures/mocked_panda.py

```diff
@@ -0,0 +1,539 @@
+import asyncio
+import logging
+import sys
+from contextlib import contextmanager
+from dataclasses import asdict, is_dataclass
+from io import BufferedReader
+from itertools import chain, repeat
+from logging import handlers
+from multiprocessing import Queue, get_context
+from multiprocessing.connection import Connection
+from pathlib import Path
+from typing import Any, Generator, Iterator, Optional, Tuple, TypeVar
+from uuid import uuid4
+
+import pytest
+import pytest_asyncio
+from aioca import purge_channel_caches
+from mock import MagicMock, patch
+from pandablocks.commands import (
+    Arm,
+    ChangeGroup,
+    Command,
+    Disarm,
+    GetBlockInfo,
+    GetChanges,
+    GetFieldInfo,
+    GetLine,
+    Put,
+)
+from pandablocks.connections import DataConnection
+from pandablocks.responses import (
+    BitMuxFieldInfo,
+    BlockInfo,
+    Changes,
+    EnumFieldInfo,
+    TimeFieldInfo,
+)
+from softioc.builder import ClearRecords
+from softioc.device_core import RecordLookup
+
+from pandablocks_ioc import create_softioc
+from pandablocks_ioc._types import EpicsName
+from pandablocks_ioc.ioc import _TimeRecordUpdater
+
+T = TypeVar("T")
+
+
+# If the test is cancelled half way through then the softioc process isn't always killed
+# Use the unique TEST_PREFIX to ensure this isn't a problem for future tests
+TEST_PREFIX = "TEST-PREFIX-" + str(uuid4())[:4].upper()
+BOBFILE_DIR = Path(__file__).parent.parent / "test-bobfiles"
+TIMEOUT = 10
+
+
+@pytest_asyncio.fixture
+def mocked_time_record_updater():
+    """An instance of _TimeRecordUpdater with MagicMocks and some default values"""
+    base_record = MagicMock()
+    base_record.name = TEST_PREFIX + ":BASE:RECORD"
+
+    # We don't have AsyncMock in Python3.7, so do it ourselves
+    client = MagicMock()
+    dummy_loop = asyncio.BaseEventLoop()
```

Comment:
Without a `loop=`, python 3.10 will throw the error `DeprecationWarning: There is no current event loop`, this doesn't happen on 3.9 and 3.11 strangely.